rotkehlchen/chain/zksync_lite/manager.py

```diff
@@ -494,34 +494,37 @@ def get_db_transactions(
         """Gets any zksynclite transactions from the DB depending on the given filter"""
         transactions = []
         with self.database.conn.read_ctx() as cursor:
-            cursor.execute(
```

Comment:
another approach that you can take is to use two cursors:

```
self.database.conn.read_ctx() as transactions_read_cursor,
self.database.conn.read_ctx() as swaps_read_cursor
```

and use both of them in different places. Can you measure the difference? I believe creating one extra cursor is not expensive and you avoid storing in memory `swap_data`. Also probably the query is more performant than using the IN operator with the subquery

Reply:
As mentioned in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1297916464845557811), doing a query per transaction would have an significant hit on performance.